src/parser/mage/frost/modules/features/WaterElemental.js

```diff
@@ -0,0 +1,199 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import SpellIcon from 'common/SpellIcon';
+import Icon from 'common/Icon';
+import { formatPercentage, formatNumber, formatThousands, formatDuration } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'interface/others/StatisticBox';
+import Analyzer from 'parser/core/Analyzer';
+
+import GlobalCooldown from 'parser/shared/modules/GlobalCooldown';
+import Abilities from 'parser/shared/modules/Abilities';
+import Channeling from 'parser/shared/modules/Channeling';
+import Haste from 'parser/shared/modules/Haste';
+import AlwaysBeCasting from './AlwaysBeCasting';
+
+class WaterElemental extends Analyzer {
+  static dependencies = {
+    haste: Haste,
+    abilities: Abilities,
+    globalCooldown: GlobalCooldown, // triggers the globalcooldown event
```

Comment:
Ah ok, initially I thought they were needed for some *magic* calculations during Bloodlust/TW maybe because it was included in the source I took them from and they are not used there too (parser/shared/modules/AlwaysBeCasting.js), now tested both and the numbers do stay the same